### ISSUE TITLE
Future vision for instant upload

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/android#553]
When we look at iOS with iPhoto and iCloud for example, it becomes very clear that instant upload is basically a hack to one-way push the photos from your mobile phone onto your computer.

But what happens if you modify them on the computer?

What you really want is simple two-way-sync. Hook up the pictures once and then it’s just all synced. How can we achieve that @nextcloud/android? Anything with the new instant upload implementation which helps us on that front?

(cc @xMartin cause we talked about that)

**Comment by AndyScherzinger** [comment]
No implementation done yet, but two way sync is part of the auto upload Roadmap #285

**Comment by eppfel** [comment]
So, you want to sync the photos back into the Android file system (outside of the internal Nextcloud folder storage)? I think your Apple use case does not apply to Android, because for editing, Apple provides software on both ends, iOS and maOS, and mirror the editing capabilities.

I would be furios, if my "originals" would be overwritten, but we have to think about "new" edited photos placed in the same folder as the uploaded ones, I guess.

@AndyScherzinger and I discussed the various sync problematics and we would like to introduce a cache quota, like on Spotify. Where would be the right place to discuss the particularities... existing/new issue?

**Comment by AndyScherzinger** [comment]
I think a new issue would be best :)

**Comment by bes1002t** [comment]
@eppfel why not let the user decide what he want? I think every user should configure which sync type he prefers.

**Comment by strugee** [comment]
@bes1002t adding new options isn't always the answer. It'd be better if we could figure out a solution for everyone without adding more settings.

Not saying that's the right way here, just that the knee-jerk "make it configurable" isn't always the best.

**Comment by bes1002t** [comment]
@strugee yes you're totally right, it's not the solution to every problem. But everybody prefers a different  way to sync their information. I think it's not userfriendly to say "do it like we want!"

**Comment by Bugsbane** [comment]
Neither is it user frierndly to have to wade through 1000 options to get the same result that most other people want.
